base/set.jl

```diff
@@ -29,14 +29,9 @@ similar(s::Set{T}) where {T} = Set{T}()
 similar(s::Set, T::Type) = Set{T}()
 
 function show(io::IO, s::Set)
-    print(io, "Set")
-    if isempty(s)
-        print(io, "{", eltype(s), "}()")
-        return
-    end
-    print(io, "(")
-    show_vector(io, s, "[", "]")
-    print(io, ")")
+    print(io, "Set(")
+    show_vector(io, s)
```

Comment:
Is there a reason why we have `print_array` but `show_vector`?

Reply:
I'm not good at naming, and I forged them on top of what was existing, but the `print_array` work for any array, while `show_vector` is a special method implemented only for 1-D collections, which is meant (as this example show) to be used beyond the array case. I just didn't change the original name for this one, and I can't think of a clearly better one.

Reply:
I mean that the discrepancy between "show" and "print" is weird, unless there's a reason for it.

Reply:
You mean in those two lines? there I just kept it how it was, and `show`ing a `String` will print the quotes. But you refered to `print_array`, which is just an internal name basically implementing the logic of `display`, so it's not so important (`print_array` comes from the pre-existing `print_matrix` function). Sorry I'm still not clear exactly what you think should be changed!

Reply:
OK, I thought you had renamed one of them. Carry on...